src/lib.rs

```diff
@@ -652,57 +652,62 @@ where
             split_words.collect::<Vec<_>>()
         };
 
+        if broken_words.is_empty() {
+            lines.push(Cow::Borrowed(""));
+            continue;
+        }
+
         #[rustfmt::skip]
         let line_lengths = |i| if i == 0 { initial_width } else { subsequent_width };
+        // This currently isn't as efficient as it could be as we collect into a vec even when it
+        // isn't necessary.
```

Comment:
Do you think it will be possible to reconcile this somehow so that both wrapping functions return the same type of iterator? If not, then it seems we lose the efficiency here?

Reply:
I want to later introduce an enum iterator that can be `WrapFirstFit` or `WrapOptimalFit`, which will gain back the efficiency hopefully. But I want to keep that in a separate PR.